integration-tests/multi_sig/multi_sig.spec.ts

```diff
@@ -0,0 +1,489 @@
+        /// 4.1 create customer and merchant multiSig session
+        ///     noticed that session state keeps updated in the Builder instance
+        customerMultiSigBuilder.createNewSession(rawTxId);
+        merchantMultiSigBuilder.createNewSession(rawTxId);
+
+        /// 4.2 generate nonce commitment
+        let customerNC = customerMultiSigBuilder.generateNonceCommitment();
+        let merchantNC = merchantMultiSigBuilder.generateNonceCommitment();
+
+        /// 4.3 add nonce commitment
+        customerMultiSigBuilder.addNonceCommitment(
+            merchant.transferKeyPair.publicKey!,
+            merchantNC,
+        );
+        merchantMultiSigBuilder.addNonceCommitment(
+            customer.transferKeyPair.publicKey,
+            customerNC,
+        );
+
+        /// 4.4 generate nonce
+        let customerNonce = customerMultiSigBuilder.generateNonce();
+        let merchantNonce = merchantMultiSigBuilder.generateNonce();
+
+        /// 4.5 add nonce
+        customerMultiSigBuilder.addNonce(
+            merchant.transferKeyPair.publicKey!,
+            merchantNonce,
+        );
+        merchantMultiSigBuilder.addNonce(
+            customer.transferKeyPair.publicKey,
+            customerNonce,
+        );
+
+        /// 4.6 partial sign
+        let customerPS = customerMultiSigBuilder.partialSign();
+        let merchantPS = merchantMultiSigBuilder.partialSign();
+
+        /// 4.7 add partial signature
+        customerMultiSigBuilder.addPartialSignature(
+            merchant.transferKeyPair.publicKey!,
+            merchantPS,
+        );
+        merchantMultiSigBuilder.addPartialSignature(
+            customer.transferKeyPair.publicKey,
+            customerPS,
+        );
+
+        const customerSig = customerMultiSigBuilder.sign();
+        const merchantSig = merchantMultiSigBuilder.sign();
+
+        // two final signature should be the same
+        expect(customerSig).to.equal(merchantSig);
+
+        // broadcast the signed tx
+        try {
+            await tendermintRpc.broadcastTx(customerSig!.toString('hex'));
+        } catch (error) {
+            console.error('ERR broadcast the final signature:', error);
+        }
+        console.log( '[LOG] complete normal flow of 2 out of 3 multiSig successfully');
+    });
+});
+
+/**
+ * Create raw tx for multiSig flow of sending back funds to different parties
+ * @param customerTxId txId of customer sending fund to multiSig addr
+ * @param firstParty one of the party base on senario
+ * @param firstPartyAmount the amount of fund getting back for the 1st party
+ * @param secondParty the other party base on senario, normally secondParty is customer
+ * @param secondPartyAmount the amount of fund getting back for the 2nd party
+ * @param viewKeys viewkeys of all parties including escrow
+ * @return txId string of raw Tx ID that waiting for signing by involved signers
+ */
+const createRawTxForMultiSig = async (
```

Comment:
@calvinlauco , I also wrap a create raw tx function here with `transaction_createRaw` RPC, but can we implement this natively?